### PR TITLE
Bug 1111417 api demos tv

### PR DESCRIPTION
**Pull request by huchengtw-moz**

### CODE REVIEW

**Comment by try-server-hook** [comment]
[huchengtw-moz John Hu (huchengtw-moz)](https://github.com/huchengtw-moz) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=e25928a3d66a)

File: tv_apps/api-demos-tv/manifest.webapp [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    "en-US": {
+      "name": "API Demos",
+      "description": "Demo app to show all APIs in TV"
+    }
+  },
+  "default_locale": "en-US",
+  "icons": {
+    "354": "/style/icons/developer.png"
+  }
+}
```
Comment (iwahashikentarou) [inline tv_apps/api-demos-tv/manifest.webapp]: Does not app start.

[1].case of "use notification web api":
- I think the following in manifest.webapp is missing.
  "messages": [
    { "notification": "/index.html" }
  ],
- This reason is due to the following code.
 gecko/b2g/components/AlertsHelper.jsm:L197
      listener.target = getNotificationURLFor(manifest.messages);

 gecko/b2g/components/AlertsHelper.jsm:L155
          gSystemMessenger.sendMessage(kNotificationSystemMessageName, {
              clicked: (detail.type === kDesktopNotificationClick),
              title: listener.title,
              body: listener.text,
              imageURL: listener.imageURL,
              lang: listener.lang,
              dir: listener.dir,
              id: listener.id,
              tag: listener.tag,
              timestamp: listener.timestamp,
              data: listener.dataObj
            },
            Services.io.newURI(listener.target, null, null),   //L155
            Services.io.newURI(listener.manifestURL, null, null)
          );


[2].case of "use IAC":
- I think something of "connect" in manifest.webapp is missing.
- Would not be a problem the following?
　 In the event handler for app.launch(), "notification" exists, but "connect" does not exist.

Comment (huchengtw-moz) [inline tv_apps/api-demos-tv/manifest.webapp]: Yes, we miss  "messages": [ { "notification": "/index.html" } ] in the first case. I will add it back.

For the [2], the connect is not missing. We do not support the app is killed after the notification shown. Once we changed the notification API to support multiple buttons. The IAC parts will be removed. And all of this parts go back to desktop notification case.

File: tv_apps/api-demos-tv/index.html [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        <p></p><p class="toolbar-title">API Tests</p><p></p>
+      </div>
+      <ul class="list">
+        <li><a href="#test=API/interactive_notification">Interactive Notification</a></li>
+      </ul>
+    </div>
+    <div id="tabs" class="tabs">
+      <ul role="tablist" class="bb-tablist" data-items="1">
+        <li id="panel2" role="presentation">
+          <a href="#API" id="API" role="tab" aria-controls="APIpanel">API</a>
```
Comment (rexboy7) [inline tv_apps/api-demos-tv/index.html]: I'm not quite familiar with `aria-` properties... but isn't it plenty with just this `<a>` tag?

File: tv_apps/api-demos-tv/index.html [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  <meta charset="utf-8">
+  <title>UI tests</title>
+  <!-- We don't need localization in this app -->
+  <link rel="stylesheet" type="text/css" href="style/uitest.css">
+  <link rel="stylesheet" type="text/css" href="style/tabs.css" >
+  <script type="text/javascript" src="js/uitest.js"></script>
+  <script defer src="shared/js/lazy_loader.js"></script>
+  <script defer src="shared/js/accessibility_helper.js"></script>
+</head>
+<body>
```
Comment (rexboy7) [inline tv_apps/api-demos-tv/index.html]: Can we lock the `x-scrollable` to `<body>`?

Comment (huchengtw-moz) [inline tv_apps/api-demos-tv/index.html]: It should be. But this version is copied from UItest. This version doesn't support keyboard operation. I will file a bug to convert the this version to support keyboard.

**Comment by autolander** [comment]
Autolander could not find a bug number in your pull request title. All pull requests should be in the format of: Bug [number] - [description].

**Comment by try-server-hook** [comment]
[huchengtw-moz John Hu (huchengtw-moz)](https://github.com/huchengtw-moz) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=f385e61df983)

**Comment by autolander** [comment]
Autolander could not find a bug number in your pull request title. All pull requests should be in the format of: Bug [number] - [description].

**Comment by try-server-hook** [comment]
[huchengtw-moz John Hu (huchengtw-moz)](https://github.com/huchengtw-moz) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=0b65fb97d888)

**Pull request closed without merging**